### ISSUE TITLE
2008  generic  types

### ISSUE THREAD

**Issue by sckoarn** [ghdl/ghdl#2382]
**Description**
6.5.3 Interface type declarations

**Expected behaviour**
simulation should run

**How to reproduce?**
run the following code

```
library ieee;
use ieee.std_logic_1164.all;

entity my_entity i
  generic (type my_type);
  port (in1 : in my_type;
  out1 : out my_type);
end entity my_entity;

architecture bot of my_entity is
begin
  out1  <= in1;
end bot;

library ieee;
use ieee.std_logic_1164.all;
use work.all;

entity top is
end entity;

architecture tb of top is
  signal a : std_logic;
  signal b : std_logic;
  signal c : std_logic;
  signal d : std_logic;

  signal g : std_logic;
  signal h : bit;

begin

  my_inst1 : entity work.my_entity(bot)
    generic map (my_type => std_logic)
    port map(in1 => a, out1 => b);

  process
  begin
    for i in 0 to 10 loop
      g <= '1';
      a <= '0';
      wait for 15 ns;
      a <= '1';
      wait for 15 ns;
    end loop;
    wait;
  end process;
  
  process
  begin
    wait for 3 ns;
    report std_logic'image(a) & " < in:out > " & std_logic'image(b) & " cout ";
    wait on b;
  end process;
end tb;

```

```
rm work-obj08.cf
ghdl -a --std=08 test.vhdl
ghdl -a --std=08 bench.vhdl
ghdl -r --std=08 work.top
```

> NOTE: `:file:` and `:image:` identifiers are specific to [issue-runner](https://github.com/1138-4EB/issue-runner). We suggest to use these, since it allows continuous integration workflows to automatically test the MWE. Using `ghdl/ghdl:*` docker images to run the MWEs ensures that the latest available GHDL is used.

> NOTE: Large files can be uploaded one-by-one or in a tarball/zipfile.

**Context**
Please, provide the following information:

- OS:
- Origin:
  - [ ] Package manager: `version`
  - [ ] Released binaries: `tarball_url`
  - [ ] Built from sources: `commit SHA`

If a `GHDL Bug occurred` block is shown in the log, please paste it here:

```
******************** GHDL Bug occurred ***************************
Please report this bug on https://github.com/ghdl/ghdl/issues
GHDL release: 3.0.0-dev (2.0.0.r1414.gbbde267d6.dirty) [Dunoon edition]
Compiled with GNAT Version: 8.3.0
Target: x86_64-linux-gnu
test/dsim/vh_tests/spec_7_08/comp17/
Command line:
ghdl -r --std=08 work.top
Exception CONSTRAINT_ERROR raised
Exception information:
raised CONSTRAINT_ERROR : trans-chap4.adb:43 access check failed
Call stack traceback locations:
0x55a1a99165ca 0x55a1a991728b 0x55a1a991da58 0x55a1a997adbb 0x55a1a98dda7f 0x55a1a98dc417 0x55a1a98e9802 0x55a1a9982c83 0x55a1a98ba4d2 0x55a1a98a82bc 0x55a1a99840f0 0x55a1a953bdfd 0x7f79df9ee099 0x55a1a953a2b8 0xfffffffffffffffe

```

**Additional context**
Add any other context about the problem here. If applicable, add screenshots to help explain your problem.

**Comment by umarcor** [comment]
If you can provide a reproducer, please open another issue.
